tests/api_connexion/endpoints/test_log_endpoint.py

```diff
@@ -14,25 +14,259 @@
+        settings_file = os.path.join(self.settings_folder, "airflow_local_settings.py")
+        new_logging_file = "LOGGING_CONFIG = {}".format(logging_config)
+        with open(settings_file, 'w') as handle:
+            handle.writelines(new_logging_file)
+        sys.path.append(self.settings_folder)
+
+        with conf_vars({('logging', 'logging_config_class'): 'airflow_local_settings.LOGGING_CONFIG'}):
+            self.app = app.create_app(testing=True)
+            self.client = self.app.test_client()
+            settings.configure_logging()
+
+    def _prepare_db(self):
+        dagbag = self.app.dag_bag  # pylint: disable=no-member
+        dag = DAG(self.DAG_ID, start_date=timezone.parse(self.default_time))
+        dag.sync_to_db()
+        dagbag.bag_dag(dag, parent_dag=dag, root_dag=dag)
+        with create_session() as session:
+            self.ti = TaskInstance(
+                task=DummyOperator(task_id=self.TASK_ID, dag=dag),
+                execution_date=timezone.parse(self.default_time)
+            )
+            self.ti.try_number = 1
+            session.merge(self.ti)
+
+    def _prepare_log_files(self):
+        dir_path = f"{self.log_dir}/{self.DAG_ID}/{self.TASK_ID}/" \
+                   f"{self.default_time.replace(':', '.')}/"
+        os.makedirs(dir_path)
+        with open(f"{dir_path}/1.log", "w+") as file:
+            file.write("Log for testing.")
+            file.flush()
+
+    def tearDown(self):
+        logging.config.dictConfig(DEFAULT_LOGGING_CONFIG)
+        clear_db_runs()
+
+        # Remove any new modules imported during the test run. This lets us
+        # import the same source files for more than one test.
+        for mod in [m for m in sys.modules if m not in self.old_modules]:
+            del sys.modules[mod]
+
+        sys.path.remove(self.settings_folder)
+        shutil.rmtree(self.settings_folder)
+        shutil.rmtree(self.log_dir)
+
+        super().tearDown()
+
+    @provide_session
+    def test_should_response_200_json(self, session):
+        self._create_dagrun(session)
+        key = self.app.config["SECRET_KEY"]
+        serializer = URLSafeSerializer(key)
+        token = serializer.dumps({"download_logs": False})
+        headers = {'Accept': 'application/json'}
+        response = self.client.get(
+            f"api/v1/dags/{self.DAG_ID}/dagRuns/TEST_DAG_RUN_ID/"
+            f"taskInstances/{self.TASK_ID}/logs/1?token={token}",
+            headers=headers
+        )
+        expected_filename = "{}/{}/{}/{}/1.log".format(
+            self.log_dir,
+            self.DAG_ID,
+            self.TASK_ID,
+            self.default_time.replace(":", ".")
+        )
+        self.assertEqual(
+            response.json['content'],
+            f"*** Reading local file: {expected_filename}\nLog for testing."
+        )
+        self.assertIn(
+            response.json['continuation_token'],
+            "{'end_of_log': True}"
```

Comment:
The token is encrypted metadata so that the user cannot manipulate it and affect its content.